### ISSUE TITLE
IInt8EntropyCalibrator2 is difficult to use

### ISSUE THREAD

**Issue by lpaatero** [bytedeco/javacpp-presets#902]
Tensorrt offers class IInt8EntropyCalibrator2 for providing scale calibration data (batches) for model conversion to 8-bit.
This class has 4 pure virtual methods getBatchSize, getBatch, readCalibrationCache and writeCalibrationCache; Purpose of class is to allow custom implementation of these methods.

In order to be useful, class org.bytedeco.tensorrt.nvinfer.IInt8EntropyCalibrator2 should be possible to extend and implement above  methods in Java.
To me it seems doing this is quite difficult. The constructor public IInt8EntropyCalibrator2(Pointer p) expects native object, so extending still requires use of javacpp (not just pure java).

Would it be possible to have org.bytedeco.tensorrt.nvinfer.IInt8EntropyCalibrator2 in such way that it has contructor public IInt8EntropyCalibrator2() and abstract java methods getBatchSize and getBatch; and native code would route any c++ calls to these methods into Java?

**Comment by saudet** [comment]
That would happen if it were "virtualized", but it's currently not tagged that way. While we're at it, could you give me a list of classes that are meant to be subclassed by users this way? Thanks!

**Comment by lpaatero** [comment]
Good question, there seems to be quite many of those. Reading the .h file, I think below is quite complete set of classes that application is likely to subclass.

IInt8Calibrator related classes:
IInt8EntropyCalibrator IInt8EntropyCalibrator2 IInt8MinMaxCalibrator IInt8LegacyCalibrator

In case someone wants to implement pure java custom layer:
IPluginLayer IPluginV2Layer 

and some classes I have not yet used: 
IOutputDimensionsFormula IAlgorithmIOInfo IAlgorithmVariant IAlgorithmContext IAlgorithm IAlgorithmSelector

**Comment by saudet** [comment]
> IInt8Calibrator related classes:
> IInt8EntropyCalibrator IInt8EntropyCalibrator2 IInt8MinMaxCalibrator IInt8LegacyCalibrator

Ok, I've added those to the last commit. Please try it with the snapshots after the builds are done in a day or two:
http://bytedeco.org/builds/

> In case someone wants to implement pure java custom layer:
> IPluginLayer IPluginV2Layer
> IPluginV2 IPluginV2Ext IPluginV2IOExt
> IPluginCreator
> 
> and some classes I have not yet used, but seems relevant:
> IOutputDimensionsFormula IAlgorithmIOInfo IAlgorithmVariant IAlgorithmContext IAlgorithm IAlgorithmSelector
> IGpuAllocator

Everything from that list, expect maybe `IPluginCreator`, seems to be either deprecated or mentions the following in comment:
```
//! \warning Do not inherit from this class, as doing so will break forward-compatibility of the API and ABI.
```
It looks like the only other classes that should be implemented are things related to plugins like `IPlugin`. I'm not sure if it makes sense to implement plugins in Java...? Let's come back to that later if it turns out to be useful for something...

**Comment by lpaatero** [comment]
Oops, the overriding warning was not in the h files.

Non-deprecated classes for custom plugin are then only IPluginV2Ext, IPluginV2IOExt and IPluginCreator.
Other non-deprecated are IAlgorithmSelector and IGpuAllocator.

**Comment by saudet** [comment]
Changes released with version 1.5.4. Thanks for reporting and please let me know if you need anything else fixed!
